src/dbnode/storage/namespace.go

```diff
@@ -1211,6 +1215,9 @@ func (n *dbNamespace) ColdFlush(flushPersist persist.FlushPreparer) error {
 	if onColdFlushDone != nil {
 		multiErr = multiErr.Add(onColdFlushDone())
 	}
+	for _, shardColdFlush := range shardColdFlushes {
+		multiErr = multiErr.Add(shardColdFlush.Done())
```

Comment:
Need to make sure we only call this if no other errors yeah?

i.e.
```
for _, shardColdFlush := range shardColdFlushes {
  if multiErr.NumErrors() != 0 {
    continue
  }
  multiErr = multiErr.Add(shardColdFlush.Done())
}
```

Probably want to do the same with evicting/rotating in-memory index results:
```go
if multiErr.NumErrors() == 0 {
  if err := onColdFlushNs.Done(); err != nil {
    multiErr = multiErr.Add(err)
  }
}
if multiErr.NumErrors() == 0 && onColdFlushDone != nil  {
  multiErr = multiErr.Add(onColdFlushDone())
}
```